From 9d989a3eb0f1d38e73013f3e866f30e5b5606ddd Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Thu, 2 Nov 2023 10:38:51 -0400
Subject: [PATCH 43/68] v4l2codecs: encoder: No need to set num_planes of plane
 size

This is going to be filled by the driver.
Upstream-Status: Pending
---
 sys/v4l2codecs/gstv4l2encoder.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/sys/v4l2codecs/gstv4l2encoder.c b/sys/v4l2codecs/gstv4l2encoder.c
index 26d79c1..0938bb1 100644
--- a/sys/v4l2codecs/gstv4l2encoder.c
+++ b/sys/v4l2codecs/gstv4l2encoder.c
@@ -622,12 +622,11 @@ gst_v4l2_encoder_set_src_fmt (GstV4l2Encoder * self, GstVideoInfo * info,
     GST_DEBUG_OBJECT (self,
         "Trying to use peer format: %" GST_FOURCC_FORMAT " %ix%i",
         GST_FOURCC_ARGS (pix_fmt), width, height);
+
     fmt.fmt.pix_mp.pixelformat = pix_fmt;
     fmt.fmt.pix_mp.width = width;
     fmt.fmt.pix_mp.height = height;
-    fmt.fmt.pix_mp.num_planes = 1;
-    fmt.fmt.pix_mp.plane_fmt[0].sizeimage =
-        GST_ROUND_UP_16 (width) * GST_ROUND_UP_16 (height) * 3 / 2;
+
     ret = ioctl (self->video_fd, VIDIOC_S_FMT, &fmt);
     if (ret < 0) {
       GST_ERROR_OBJECT (self, "VIDIOC_S_FMT failed: %s", g_strerror (errno));
-- 
2.25.1

